sysl2/sysl/codegen.go

```diff
@@ -231,12 +231,27 @@ func loadAndGetDefaultApp(root, model string) (*sysl.Module, string) {
 
 // GenerateCode transform input sysl model to code in the target language described by
 // grammar and a sysl transform
-func GenerateCode(rootModel, model, rootTransform, transform, grammar, start string) []*CodeGenOutput {
+func GenerateCode(codegenParams *CmdContextParamCodegen) []*CodeGenOutput {
 	var codeOutput []*CodeGenOutput
 
-	mod, modelAppName := loadAndGetDefaultApp(rootModel, model)
-	tx, transformAppName := loadAndGetDefaultApp(rootTransform, transform)
-	g := readGrammar(grammar, "gen", start)
+	logrus.Warnf("root-model: %s\n", *codegenParams.rootModel)
+	logrus.Warnf("root-transform: %s\n", *codegenParams.rootTransform)
+	logrus.Warnf("model: %s\n", *codegenParams.model)
+	logrus.Warnf("transform: %s\n", *codegenParams.transform)
+	logrus.Warnf("grammar: %s\n", *codegenParams.grammar)
+	logrus.Warnf("start: %s\n", *codegenParams.start)
+	logrus.Warnf("loglevel: %s\n", *codegenParams.loglevel)
+
+	if *codegenParams.verbose {
+		*codegenParams.loglevel = debug
+	}
+	if level, has := defaultLevel[*codegenParams.loglevel]; has {
```

Comment:
What would be the log level if `!has`?

Reply:
It will be defaulted to "warn", if there is no input in the command line.

Reply:
Cool. May be mention that in a comment.

Reply:
Default logging changed to info.